run_freyja.sh

```diff
@@ -0,0 +1,109 @@
+
+for bam in ${in_dir}/*_trim.sorted.bam; 
+do 
+    infile=${bam##*/}; echo "$(date): input file name is $infile"
+    sample=${infile%-UT*}; echo "$(date): base name is $sample" 
+ 
+    singularity exec --bind ${analysis_dir} uphl-freyja-latest.simg freyja variants ${bam} --variants $workdir/${sample}_out_variants --depths $workdir/${sample}_out_depths --ref ${scov2}
+
+    depth=${sample}_out_depths
+    var=${sample}_out_variants.tsv
+
+    #depth_path="$workdir/${var}" 
+    #var_path="$workdir/${depth}"
+
+    #output_file="$outdir${sample}_lin_out.tsv"
+    echo "$(date): Running Demultiplexing step for $sample" >> $status_file
+    singularity exec --bind ${analysis_dir} uphl-freyja-latest.simg freyja demix $workdir/${var} $workdir/${depth} --eps 0.01 --covcut 10 --confirmedonly --output $outdir${sample}_lin_out.tsv
+
+    #echo "$(date): Boostrap analysis step running for $sample" | tee -a $log_file
+    #freyja boot $workdir/${var} $workdir/${depth} --nt 8 --nb 500 --eps 0.01 --output_base $outdir${sample}_boot.tsv
+
+    # Check the exit status of the 'freyja demix' command. This is useful for tracking which samples failed at the demultiplex stage.
+    if [ $? -eq 0 ]; then
+        if [ -f "${outdir}/${sample}_lin_out.tsv" ]; then
+        echo "Run $run_name: Sample $sample processed successfully." >> $status_file
+        fi
+    else
+        echo "Run $run_name: Sample $sample did not process successfully. SolverError: Solver 'ECOS' failed. Skipping this sample." >> $status_file
+    fi
+
+done
+
+echo "$(date): Demultiplexing step completed and results are stored in $outdir" >> $status_file
+echo "$(date): Running lineage aggregation step in Freyja"
+
+singularity exec --bind ${analysis_dir} uphl-freyja-latest.simg freyja aggregate $outdir --output $outdir${run_name}_lineages_aggregate.tsv --ext tsv
+
+#echo "$(date): Plotting lineage aggregate output from Frejya"
+#singularity exec --bind ${analysis_dir} uphl-freyja-latest.simg freyja plot ${outdir}${run_name}_lineages_aggregate.tsv --output $outdir${run_name}_lineages_aggregate_plot.png --lineages
+
+# Check if Freyja analysis completed successfully and the aggregate output file is generated
+if [ -f "${outdir}${run_name}_lineages_aggregate.tsv" ]; then
+    echo "$(date): Lineage aggregration completed and aggregrated tsv is stored in $outdir. Freyja analysis completed"
+else
+    echo "$(date): Lineage aggregration failed. Please check the input files and try again."
+    exit 1
+fi
+
+echo "$(date) : Copying Freyja aggregate lineage results to $results"
+#cd $outdir
+cp $outdir*lineages_aggregate.tsv $results
+
+echo "$(date): Extracting freyja lineage dictionary results and converting it to a long dataframe for downstream processing using the python script freyja_custom_lin_processing.py"
+/home/linuxbrew/.linuxbrew/bin/python3 ${freyja_cln_tsv} ${run_name}
```

Comment:
I think this line is missing something. Should it be

```
/home/linuxbrew/.linuxbrew/bin/python3 <path to >/freyja_custom_lin_processing.py ${freyja_cln_tsv} ${run_name}
```

Reply:
I have the <path to >/freyja_custom_lin_processing.py defined on Line 26.

Reply:
Ah! freyja_cln_tsv is the path to the script. My apologies!!!